inference-engine/src/gna_plugin/optimizer/gna_pass_manager.cpp

```diff
@@ -654,7 +669,7 @@ void RemovePermutationsNHWCToNCHWPass::run() {
         auto next = getInputTo(l->outData.front()).begin()->second;
         auto prev = CNNNetPrevLayer(l);
 
-        if (!LayerInfo(next).isPermute() || !LayerInfo(prev).isPermute()) {
+        if (!isPermuteToRemove(next, false) || !isPermuteToRemove(prev, true)) {
```

Comment:
rename to more correct like canBeRemoved, since you are dealing with permute and reshape as i can see - while this better to be explicitly formulated in name  for example identityPermute or other 
Reshape layers matter since they allow correct dims to be present in input and outputs of functional layer - this is why it is no recommended to remove the,